File: 151003/Mitrofanov/Publisher/Entity/Db/Editor.cs
using Publisher.Entity.Common;
using System.ComponentModel.DataAnnotations;

namespace Publisher.Entity.Db
{
    public class Editor(string login, string password, string firstName, string lastName) : AbstractEntity
    {
        [MinLength(2)]
        public string Login { get; set; } = login;
        [MinLength(8)]
        public string Password { get; set; } = password;
        [MinLength(2)]
        public string FirstName { get; set; } = firstName;
        [MinLength(2)]
        public string LastName { get; set; } = lastName;
        public ICollection<Story> Storys { get; set; } = [];


        public Editor(int id, string login, string password, string firstName, string lastName) :
            this(login, password, firstName, lastName)
        {
            Id = id;
        }

        public Editor() : this(string.Empty, string.Empty, string.Empty, string.Empty) { }
    }
}

File: 151001/Nesteruk/task1/src/main/java/by/harlap/rest/model/Editor.java
package by.harlap.rest.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class Editor extends AbstractEntity {

    private String login;
    private String password;
    private String firstname;
    private String lastname;
}
